### ISSUE TITLE
Removing known SCAM coins and accounts

### ISSUE THREAD

**Issue by sschiessl-bcp** [bitshares/bitshares-ui#1634]
**Is your feature request related to a problem? Please describe.**
The UI is already highlighting suspected scam accounts
![image](https://user-images.githubusercontent.com/33128181/42088170-f115514a-7b98-11e8-9348-e34fbb4da17e.png)
and it should include the same feature for assets. I haven't checked the actual implementation there yet but I envision to have single source of truth for both scammer lists and a issue for every single account / scam coin that will contain papertrail as to why it was put on the list. This will give for defense if necessary and also enough searchable history for the future.

Example for potential scam: 
- https://wallet.bitshares.org/#/asset/EOS
- or anything that has 100% market fee
- accounts involved in OL hack etc.

**Comment by sschiessl-bcp** [comment]
@bitshares/ui-dev

**Comment by happyconcepts** [comment]
Before you stigmatize all 100% market fee coins I would ask you to at least show us an example of where there is a problem. 

A few of my clients are at MVP with flat fee business models on bitshares, and it requires setting 100% to do that. So what you propose basically pulls the rug out from under those potential businesses.

**Comment by wmbutler** [comment]
@happyconcepts can you further explain a "flat fee business model"?

**Comment by fractalnode** [comment]
I propose to approach this from a different angle
```
Deny all, 
Allow from (...)
```
https://github.com/bitshares/bitshares-core/issues/1044

**Comment by sschiessl-bcp** [comment]
Either way, sorry for doing generalizations.

 The idea is that every coin/account that would be considered receives an issue that allows discussion. Your use case would of course be heard @happyconcepts.

We don't need to discuss this in here, this issue is to create a general framework for it.

**Comment by abitmore** [comment]
@wmbutler: it's a smart idea. Because order creation fee is much lower than transfer fee, it can be (ab)used by businesses as payment method. By setting a 100% market fee, it's guarantees that nobody else can create another order on the market competing with the asset issuer.

**Comment by wmbutler** [comment]
> it can be (ab)used by businesses

You are saying that the asset can be abused by businesses as a payment method? You mean arbitrary businesses or the business that created the asset?

> it's guarantees that nobody else can create another order on the market competing with the asset issuer

Hrmmm, I thought it was the nature of markets to compete. I'm not trying to argumentative. Just trying to understand the various use cases and how we can resolve them.

In these cases, are the issuers of the asset electing not to use the exchange at all and just use the asset for sending and receiving?

**Comment by happyconcepts** [comment]
> @happyconcepts can you further explain a "flat fee business model"?

So a classic example would be the network fee to place buy orders via `limit order create`. The same fee amount (e.g. flat) is applied regardless of the amount of the order. 

Hey @wmbutler @sschiessl-bcp @svk31 on a side note can you review #1529 with me to reinstate the agreed, original [2] bounty? These browser timezone issues are tricky at best and messy at worst, which I knew at the outset, but to have put in the time and thoroughly tested on chrome only to get knocked down to [0.5] does not feel good in principal at all... any PM help to alleviate the situation would be appreciated.

**Comment by gibbsfromncis** [comment]
@happyconcepts please move your discussions about another issue in this issue.

**Comment by svk31** [comment]
@happyconcepts You may not have noticed but as startail pointed out your PR for that issue actually caused 'invalid date' to appear in Firefox, so I fixed the issue myself in a new commit. You'll still get the 0.5 hours but please stop asking for 2 hours for a PR that in the end did not even solve the issue.

**Comment by sschiessl-bcp** [comment]
First candidate: TELETON

**Comment by abitmore** [comment]
> In these cases, are the issuers of the asset electing not to use the exchange at all and just use the asset for sending and receiving?

@wmbutler yes. They're effectively transferring funds with market creation fee rate which is much lower according to current fee schedule.

**Comment by sschiessl-bcp** [comment]
Before we get further into discussion of actual assets, this issue is to decide if we want such a feature in the UI. 

I imagine:
 - if the market fee is above a certain %, always notify the user (non invasive) to double check what he is doing
 - if the asset is on the blacklist, notify the user when transferring (invasive) , viewing balance (non-invasive) and trying to trade (invasive), additionally mark operations in Dashboard that contain said asset (non-invasive)
 - if the account is on the blacklist, notify the user when transferring to it (invasive) and mark operations in the Dashboard that contain said account (non-invasive)

**Comment by wmbutler** [comment]
@happyconcepts credited you two hours but please note @svk31 comment. @sschiessl-bcp agreed on all three. Really need UX on those three first.

**Comment by happyconcepts** [comment]
@wmbutler thank you, and noted.

> They're effectively transferring funds with market creation fee rate which is much lower according to current fee schedule

@abitmore can someone provide a real example so I can look at it and understand the problem? Who does this hurt?

Im surprised to hear of this use case, but pleasantly surprised at that. This sounds like innovation, something that bitshares needs to encourage instead of blacklisting it.

**Comment by abitmore** [comment]
> Who does this hurt?

The DAC itself. It may loss some income due to this. On the other hand, perhaps the DAC will get more adoption due to the low fee. Who knows. I'm not judging here, just analyzing.

**Comment by sschiessl-bcp** [comment]
The goal is not to eliminate all assets with 100% market fee.

But there are scam coins, there was one for Tron. Essentially the attempt is to have a similar or even the exact name as the real token and put 100% market fee. The unknown user thinks it's the actual token and buys it, ending up losing his complete investment. 

@happyconcepts what do you mean by "Im surprised to hear of this use case, but pleasantly surprised at that". In your first comment you said you are doing that for some of your clients, contracting the statement above

**Comment by happyconcepts** [comment]
Thanks for discussion and pointing out a scam example. 

This approach surprised me because both of the projects I am backing are designed to conduct transactions via the DEX just like regular trades. We set the market fee max and make the fee 100%.

So what we obviously need to avoid is presenting new users with a big warning about our 100% fee when it is not a problem at all.

Your and the Foundation's understanding and support in that regard is very much appreciated.

**Comment by sschiessl-bcp** [comment]
> Your and the Foundation's understanding and support in that regard is very much appreciated.

Of course, collecting facts and always curious about business models. Anyways, nothing I do with this account is representing the Foundation.

> Disclose the maximum fee instead of the market fee percent. That is my best initial suggestion.

The market fee is already displayed, but many users do not notice it. I'm thinnking to include another notice in the Confirm Transaction Popup.

**Comment by happyconcepts** [comment]
So when we bring a new user to Bitshares to transaction in our token, you are going to alert that user with SCAM about us.

Solely because we set market fee % parameter of the token to 100%.

Is this really the best approach for Bitshares? To squeeze out entrepreneurs like me and justifying it over paranoia about some potential scam?

**Comment by sschiessl-bcp** [comment]
A customized UI that looks like a transfer but uses a 100% market fee UIA would certainly be the best approach. 

Nevertheless @happyconcepts my concept idea would still stand. Every asset that gets a scam popup warning would be inserted via github issues, which allows discussion for every asset. A 100% market fee asset that has not been included via an issue would only receive a non-inversive warning on the height of the market fee but not a scam popup warning. (actual UX not decided or layed out yet)

**Comment by abitmore** [comment]
@happyconcepts if you're willing to help, would you please provide a constructive solution? Emotional words aren't helping.

**Comment by happyconcepts** [comment]
> Anyone navigating to the exchange page of an asset with 100% market fee should at minimum be shown a big warning explaining that if they attempt to make trades they will lose their funds.

Then do it on the confirmation page, please. For the following very important reason:

>  they will lose their funds.

Not true. Simply not true. When you buy our SPARK asset on the exchange page the most you could lose is less than $1.00 because the amount of our fee is capped by the maximum fee parameter. 

@abitmore thank you I am still trying. To re-state it, why not show the market fee in actual amount that will be charged, or at least an estimate. Show market fee **amount**, on confirm modal, as it is more relevant than the percentage.

**Comment by abitmore** [comment]
> fee is capped by the maximum fee parameter

Makes sense.

**Comment by svk31** [comment]
>> they will lose their funds.

> Not true. Simply not true. When you buy our SPARK asset on the exchange page the most you could lose is less than $1.00 because the amount of our fee is capped by the maximum fee parameter.

Sure, but the above scenario of a legit use of a 100% fee also implies that the max fee is set to a high enough value that it never comes into play. A market as you describe with 100% that is capped by a maximum fee doesn't fall into that scenario, so then the question becomes we need answered is why such a market should be treated as a legitimate market. Anyone making trades in such a market will need to understand that unless their order exceeds the maximum fee they will lose all of their funds as explained above.

**Comment by happyconcepts** [comment]
People buying our coin are to pay a market fee. A flat fee of approx $1 is what we would like to do.

Assuming the bitshares system is smart enough, will you kindly suggest, then, another way to accomplish this @svk31?

**Comment by happyconcepts** [comment]
@bytemaster greetings bitshares 3 needs flat market fee parameter thanks

**Comment by wmbutler** [comment]
@happyconcepts Why wouldn't these companies offering these assets just fork the reference wallet and display what they like?

**Comment by sschiessl-bcp** [comment]
Please stop the pointless discussion now.

This issue is for a general framework. Your discussion is for one specific asset and if it should be included in the list or not. You can do that later in the corresponding future issue.

If anyone has contributions to the actual framework on how to mark such assets please go ahead.

**Comment by happyconcepts** [comment]
> If anyone has contributions to the actual framework on how to mark such assets please go ahead.

I cannot think of a better way to destroy public confidence in bitshares than by introducing the thought of scam to as many users as prominently as possible, which is why I am really surprised that this issue is being approached this way at all. 

Thus, again, I urge you to place your scam warnings, alerts, emotions on the confirmation page and not in the OE box which as I have said before is crowded enough already among other reasons.

**Comment by happyconcepts** [comment]
> @happyconcepts Why wouldn't these companies offering these assets just fork the reference wallet and display what they like?

Do you mean like bitSPARK launching their Sendy app last week? I would presume that to be a custom front end.

But to answer your question, at Spark we want to stay with the reference UI for two main reasons, one internal and one external:

1. Because we aren't selfish. We want to introduce our users to the bitshares DEX and see them trade happily on the bitshares DEX. Seeing our users trade other coins like bitUSD and bitCNY makes us happy for our overall investment in bitshares, of course, and there is immeasurable pleasure in seeing our users return for future rounds with Spark.

2.  Externally the exchange market is fragmented enough, in my opinion. Ours is simply a different payment structure in an era of increasingly "smart coins". If we must fork and make our own wallet just for this... then why even stay on bitshares instead of going to waves or another chain?

**Comment by abitmore** [comment]
![image](https://user-images.githubusercontent.com/9946777/43038736-6ff401b8-8d1f-11e8-9800-245d3527f48a.png)

**Comment by wmbutler** [comment]
@happyconcepts are you a representative of BitSpark? I have no idea what point you are trying to prove.

**Comment by wmbutler** [comment]
@sschiessl-bcp I'd suggest you display percentages and close this issue.

**Comment by sschiessl-bcp** [comment]
I suppose that would be the way with least resistance. Will look into it

**Comment by happyconcepts** [comment]
> @happyconcepts are you a representative of BitSpark? I have no idea what point you are trying to prove.

No, but I understand George has started calling themselves spark instead of bitspark. Perhaps a future discussion will result in their use of the spark token I created here last year, but I don't appreciate being railroaded and labeled a scam here with no basis in the meantime.

I am just providing the community with a spark price tracker now. Pivoted development to waves for our flat fee businesses since I now understand from this thread that here we will be branded as scam. I have nothing to prove, really.
 
I do have some free time through August and would be happy to contribute documentation or testing if you want it.

**Comment by sschiessl-bcp** [comment]
> I don't appreciate being railroaded and labeled a scam here with no basi

Please read what I wrote, no one is labeling you. Also we"ll probably go with Bill's approach.

> Pivoted development to waves for our flat fee businesses

Much success on the Waves platform, we'd be happy to still have you here as well

**Comment by kapeer42** [comment]
Can I claim this?

**Comment by startailcoon** [comment]
I feel this issue needs more discussion.

@wmbutler you renamed this to *Display percentage for fees in Market Order area*. What exactly does this refere to? As we already have market fee displayed on the market, what else do we need to display?

The Discussion started with hiding accounts that was potential scammers, but the topic has changed since the start. What is the goal of this issue, what do we need to change or acomplish?

**Comment by wmbutler** [comment]
The original issue was to display accounts with high market fees. The conversation led to adding the percentage next to the actual fee to make it obvious that it was a high value.

**Comment by happyconcepts** [comment]
Sure would be nice if there were room in the repo for others.

**Comment by sschiessl-bcp** [comment]
> Sure would be nice if there were room in the repo for others.

For this issue kapeer42 is in line, I assume your comment was against startail. Can you elaborate?

**Comment by happyconcepts** [comment]
Again? Please....

**Comment by sschiessl-bcp** [comment]
> Again? Please....

I want to see if there are any ongoings I wasn't aware of

**Comment by happyconcepts** [comment]
Send me an email or unban me from telegram if you are going to confront every single comment of mine, does that sound reasonable?

**Comment by sschiessl-bcp** [comment]
> Send me an email or unban me from telegram if you are going to confront every single comment of mine, does that sound reasonable?

GitHub does not expose emails, what is yours? Please contact me in telegram @sschiessl

**Comment by happyconcepts** [comment]
Can't , banned. Convenient for the thieves, all too convenient...

**Comment by sschiessl-bcp** [comment]
What do you mean banned? If you have Telegram you can contact me personally, independent which channels you are banned. My handle is sschiessl

**Comment by happyconcepts** [comment]
I deserve venue in the group channel, as an investor, as a contributor. Gee I wonder why I am not enthused about bitshares?

**Comment by sschiessl-bcp** [comment]
I can't help you to get unbanned if I don't know your nick, and it is easier to communicate if you find me there

**Comment by wmbutler** [comment]
@happyconcepts if you were kind it would make a difference.

**Comment by happyconcepts** [comment]
There is a role on every competitive team for someone like me who isn't afraid to call bullshit on sight.

**Comment by happyconcepts** [comment]
Not in my opinion. Indeed that is a very rough paste to put the fee on the wrong line...are you sure you understand the issue?

This is the Order Entry (OE) box.  . .

It is really, really important to the future of the DEX that people can trade ... and trade quickly ... and trade accurately. So the Price, Pair, and Quantity are important. I would not marginalize them by using such small size relative to your HUGE size WARNING.

If you drive all the traders away by making it hard to use the OE or making every new trader think BitShares DEX is a scam platform by making the concept so upfront and prevalent, then I suppose it won't matter what you put in the OE or what size you want to make it.

Restaurants do not put up signs telling customers about the potential of food poisoning from their product. Yet here you want to kill the business by putting up needless signs. I don't approve, ultimately, because it's a stupid tradeoff.

**Comment by kapeer42** [comment]
@wmbutler 
Can you explain what
> adding the percentage next to the actual fee

means please?

Should it be like shown in @abitmore [comment](https://github.com/bitshares/bitshares-ui/issues/1634#issuecomment-406812752) ? This doesn't seem like a percentage next to a fee.

@happyconcepts
> put the fee on the wrong line

 Thanks for catching this, I try to extract information without spending too much time reading. My bad. I understand now that this is about market fee.
> are you sure you understand the issue?

No, hence why I ask for confirmation.
> This is the Order Entry (OE) box. . .
 
Well 'Display percentage for fees in Market Order area' led me to believe I had to add a percentage near where a market order is made.
> HUGE size WARNING

This is what I meant by rough paste, should have been clearer sorry. 
I agree this is way too big, I didn't want to loose time reopening gimp and reducing the size.
